web_src/css/repo.css

```diff
@@ -3083,46 +3083,86 @@ tbody.commit-list {
   }
 }
 
-.pr-status {
+.ui.segment.commit-status-list {
+  padding: 0;
+  max-height: 30vh;
+  overflow-x: hidden;
+  transition: max-height 0.4s;
+  border-bottom: 0 !important;
+}
+
+.pr-commit-status.hide {
+  border-bottom: 0 !important;
+}
+
+.pr-commit-status.hide .ui.attached.header {
+  border-bottom: 0 !important;
+}
+
+.pr-commit-status.hide .commit-status-list {
+  padding: 0;
+  max-height: 0;
+  overflow-x: hidden;
+  transition: max-height 0.4s;
+  border: 0;
+}
+
+.commit-status-item {
   padding: 0 !important; /* To clear fomantic's padding on .ui.segment elements */
   display: flex;
   align-items: center;
 }
 
-.pr-status .commit-status {
+.commit-status-item:first-child {
+  border-top: none !important;
+}
+
+.commit-status-item:last-child {
+  border-bottom: none !important;
+}
+
+.commit-status-item .commit-status {
   margin: 1em;
   flex-shrink: 0;
 }
 
-.pr-status .status-context {
+.commit-status-item .status-context {
   display: flex;
   justify-content: space-between;
   width: 100%;
 }
 
-.pr-status .status-context > span {
+.commit-status-item .status-context > span {
   padding: 1em 0;
 }
 
-.pr-status .status-details {
+.commit-status-item .status-details {
   display: flex;
   padding-right: 0.5em;
   align-items: center;
   justify-content: flex-end;
 }
 
 @media (max-width: 767.98px) {
-  .pr-status .status-details {
+  .commit-status-item .status-details {
     flex-direction: column;
     align-items: flex-end;
     justify-content: center;
   }
 }
 
-.pr-status .status-details > span {
+.commit-status-item .status-details > span {
   padding-right: 0.5em; /* To match the alignment with the "required" label */
 }
 
+.tippy-commit-status {
+  max-width: 100vw;
+}
```

Comment:
I think this could be a global style on all tippies to ensure they do not stretch beyond screen width. Likely leave some space as well so `max-width: calc(100vw - 16px)`.

Reply:
Done in https://github.com/go-gitea/gitea/pull/26284/commits/08f9ccd0721a3392ff1523a06c0f0f38ca901fb5.